src/lib_json/json_reader.cpp

```diff
@@ -1925,9 +1922,7 @@ bool CharReaderBuilder::validate(Json::Value* invalid) const {
   std::set<String> valid_keys;
```

Comment:
We should be able to apply the same static set technique here as in the Writer, and remove the getValidReaderKeys function.

Reply:
Done.